### PR TITLE
Add EnvWarningBanner PoC

### PR DESCRIPTION
**Pull request by minimalsm**
## Description

Proposing we remove individual warnings about `.env` safety in Tutorial files with a reusable component.

**Why?**

1. Reduces duplicate work for translators
2. More maintainable (allows us to refine component/message to increase effectiveness)
3. Can be dropped anywhere we're telling people to use `.env` files, increasing developer safety
4. Allows us to remove many HTML tags (`<code>` and `<a>` tags) across our tutorial markdown pages, improving maintenance during Crowdin import process

The current banner is used in many places, but really should be in more places to avoid people getting by poor env hygiene:

Screenshot 2023-05-04 at 14 30 29

## Related Issue

#10126

### CODE REVIEW

**Review by pettinarip — CHANGES_REQUESTED** [review]
Hey @minimalsm I think this change makes total sense but the only _big_ downside I see is in adding more text (translations) in the intl JSON files.

I'm a bit worried about the impact in memory usage on build time.

What we have been trying to encourage recently is the opposite of this PR: to set translations in md files as much as we can, and leave translations in the JSON files just for the global components.

I'd love to explore more these types of abstractions in the future but I see the previous reason as a hard blocker.

LMK what do you think.

**Review by pettinarip — APPROVED** [review]

**Review by nhsz — CHANGES_REQUESTED** [review]
Thanks @minimalsm! Asking for a few changes

File: src/components/EnvWarningBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from "react"
+import { FlexProps } from "@chakra-ui/react"
+import InfoBanner from "./InfoBanner"
+import Translation from "./Translation"
+
+export interface IProps extends FlexProps {}
```
[the remark this answers is not in the document]
Comment (minimalsm) [reply src/components/EnvWarningBanner.tsx]: Done, thanks.

File: src/intl/en/page-developers-tutorials.json [hunk 2]
```diff
@@ -24,5 +24,6 @@
   "page-tutorial-intermediate": "Intermediate",
   "page-tutorial-advanced": "Advanced",
   "page-find-wallet-try-removing": "Try removing a feature or two",
-  "page-find-wallet-clear": "Clear filters"
+  "page-find-wallet-clear": "Clear filters",
+  "page-tutorials-env-banner": "Don't commit <code>.env</code>! Please make sure never to share or expose your <code>.env</code> file with anyone, as you are compromising your secrets in doing so. If you are using version control, add your <code>.env</code> to a <a href=\"https://git-scm.com/docs/gitignore\">gitignore</a> file."
```
[the remark this answers is not in the document]
Comment (minimalsm) [reply src/intl/en/page-developers-tutorials.json:28]: I think we might get this automatically from our Link component @nhsz 🤔 

[In our templates, we use MDXProvider](https://github.com/ethereum/ethereum-org-website/blob/25ef0f8e8a1eb6a145114e30c74748a73a9e660d/src/templates/static.tsx#L171) to override the default the rendering of HTML elements.

In our Link, [we parse the href to determine if a link is internal/external](https://github.com/ethereum/ethereum-org-website/blob/25ef0f8e8a1eb6a145114e30c74748a73a9e660d/src/components/Link.tsx#L118).

Tested in dev, and looks like it is working as expected without the additional properties.

**Pull request merged**